src/pages/DeworkCallbackPage/DeworkCallbackPage.tsx

```diff
@@ -0,0 +1,59 @@
+import { FC, useCallback, useEffect, useMemo, useState } from 'react';
```

Comment:
Thinking about the callback pattern for other integrations...

So if we add Discord, and we want to have it hit a Discord callback page, is that page going to be substantially different than the Dework callback page? If quite similar, could we just switch this to be a generic integration callback page like /integration-callback?integration_name='DeWork' or something since this is just a placeholder for the UI waiting for external interaction?

Reply:
I might be missing some context here, but I think this would be better as a serverless function under `api/` rather than a page in the frontend. We've got some fairly major changes to the UI planned for the next 3-6 months, and the API is less likely to obstruct / be broken by those.

If we still want to have a bit of UI that acknowledges the success or failure of the operation, we can use `useApeSnackbar` on AdminPage to show a message after redirecting back.

Reply:
^ @topocount what do you think?

Reply:
@0xchomper the team chatted about this a bit more & we realized that my suggestion is a non-starter at this time because of how we're doing auth with bearer tokens. 

So all I'd suggest for this page now is to separate out the general loading UI from the Dework-specific logic, like @crabsinger was saying. Something like this might work (just a suggestion, feel free to do it however it makes sense to you):

```js
// rename to IntegrationCallbackPage.ts

const integrations = [
  {
    name: 'dework',
    test: params => !!params.get('dework_organization_id'),
    render: setStatus => <DeworkIntegration setStatus={setStatus} />
  },
  // future integrations...
]

// omitting a lot of details to show the basic structure

const IntegrationCallbackPage = () => {
  const integration = integrations.find(i => i.test(params));
  const [status, setStatus] = useState('loading');

  return <div>
    {status === 'loading' ? <LoadingModal /> : <BackToCircles>}
    {integration.render(setStatus)}
  </div>
}

const DeworkIntegration = ({ setStatus }) => {
  useEffect(() => {
    doWork().then(() => setStatus('success'));
  }, [])
}

Reply:
Nice - updated!